mod-di-converter-storage-server/src/main/java/org/folio/services/migration/ProfileMigrationServiceImpl.java

```diff
@@ -64,26 +64,38 @@ public Future<Boolean> migrateDataImportProfiles(Map<String, String> headers, Co
     return profileWrapperDao.checkIfDataInTableExists(tenantId)
       .compose(isDataPresent -> {
         if (!isDataPresent) {
-          LOGGER.info("Profile migration started...");
-          return runScript(tenantId, REVERT_VIEW)
-            .compose(y -> jobProfileDao.getTotalProfilesNumber(tenantId))
-            .compose(x -> jobProfileDao.getProfiles(true, true, "cql.allRecords=1   ", 0, x, tenantId))
-            .compose(f -> {
-              List<Future<List<ProfileSnapshotItem>>> snapshotList = new ArrayList<>();
-              for (JobProfile jobProfile : f.getJobProfiles()) {
-                snapshotList.add(profileSnapshotDao.getSnapshotItems(jobProfile.getId(), ProfileSnapshotWrapper.ContentType.JOB_PROFILE, jobProfile.getId(), tenantId));
-              }
-              return wrapAndCreateProfileWrappers(creationProfilesFuture, tenantId, snapshotList);
-            })
-            .compose(t -> runScript(tenantId, UPDATE_SCHEMA_FOR_MIGRATION))
-            .compose(r -> runScript(tenantId, UPDATE_WRAPPERS_INSIDE_ASSOCIATIONS));
+          return checkForInvalidProfilesAndFailMigrationIfAnyExist(tenantId)
+            .compose(p -> {
+              LOGGER.info("Profile migration started...");
+              return runScript(tenantId, REVERT_VIEW)
+                .compose(y -> jobProfileDao.getTotalProfilesNumber(tenantId))
+                .compose(x -> jobProfileDao.getProfiles(true, true, "cql.allRecords=1   ", 0, x, tenantId))
+                .compose(f -> {
+                  List<Future<List<ProfileSnapshotItem>>> snapshotList = new ArrayList<>();
+                  for (JobProfile jobProfile : f.getJobProfiles()) {
+                    snapshotList.add(profileSnapshotDao.getSnapshotItems(jobProfile.getId(), ProfileSnapshotWrapper.ContentType.JOB_PROFILE, jobProfile.getId(), tenantId));
+                  }
+                  return wrapAndCreateProfileWrappers(creationProfilesFuture, tenantId, snapshotList);
+                })
+                .compose(t -> runScript(tenantId, UPDATE_SCHEMA_FOR_MIGRATION))
+                .compose(r -> runScript(tenantId, UPDATE_WRAPPERS_INSIDE_ASSOCIATIONS));
+            });
         } else {
           LOGGER.info("Migration will not execute. profile_wrappers table is NOT empty already.");
           return Future.succeededFuture(true);
         }
       });
   }
 
+  private Future<Void> checkForInvalidProfilesAndFailMigrationIfAnyExist(String tenantId) {
```

Comment:
@yaroslav-epam do we need action_to_mapping there?

Reply:
there is no way to create a mirror association between action and mapping profiles

Reply:
@KaterynaSenchenko  Got it.Thank you!